components/forms/request-premium/index.jsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState, useCallback } from 'react'
+import { classNames } from '@oacore/design/lib/utils'
+
+import styles from './styles.module.css'
+
+import { Button, TextField } from 'design'
+
+// TODO: Should be moved to @oacore/design
+const Textarea = ({
+  id,
+  className,
+  label,
+  tag: Tag = 'p',
+  ...textareaProps
+}) => (
+  <Tag id={id} className={classNames.use(styles.textarea).join(className)}>
+    <textarea id={`${id}-textarea`} {...textareaProps} />
+    <label htmlFor={`${id}-textarea`}>{label}</label>
+  </Tag>
+)
+
+const Form = ({
+  // data props
+  fromEmail,
+  fromName,
+  subject,
+  body,
+
+  // event handlers
+  // onChange and onSubmit can be passed implicitly
+  onCancel,
+
+  // html props
+  children,
+  className,
+  ...formProps
+}) => {
+  // Showing name only when email is edited de
+  const [showName, toggleName] = useState(false)
+
+  // Showing message only when user decided to edit it
+  const [showMessage, toggleMessage] = useState(false)
+
+  const handleMessageToggle = useCallback(() => toggleMessage(true), [])
+
+  return (
+    <form
+      className={classNames.use(styles.container).join(className)}
+      {...formProps}
+    >
+      <div className={styles.personal}>
+        <TextField
+          type="email"
+          name="email"
+          defaultValue={fromEmail}
+          autoComplete="email"
+          label="Email"
+          helper="We will reach you back using this email address"
+          onChange={useCallback(() => toggleName(true), [])}
```

Comment:
I thought it should have an effect here but I am not sure. 

Working with hooks closer I find it quite interesting and tricky. Their behaviour is not obvious even if the interface is quite clear. I assume that hook call is replaced with something during compilation to prevent, for example, creating a function on every render in this case.

I should learn it more deeply to be sure in what I say.

Reply:
You are right. I tested it and it has an effect. :)

Reply:
How did you test it?